src/shared/conf-parser.c

```diff
@@ -960,3 +960,38 @@ int config_parse_ifname(
 
         return 0;
 }
+
+int config_parse_ip_port(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint16_t *s = data;
+        uint16_t port;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue))
+                return 0;
```

Comment:
hmm, i wonder what the best approach here would be. Normally we say that assigning the empty string to a setting this should have the effect of resetting things to the "default", for whatever that might be. Now, implementing this properly here is difficult, since we don't know the right default, as it might be different for each user of this function. Hoewver, I think it would make sense to assign the port "0" in this case, and then make sure that the code using this setting treats 0 as "the default" when using the field.

Long story short: please change this to: 
```c
if (isempty(rvalue)) {
        *s = 0;
        return 0;
}
```